File: backend-dotnet/Models/Initiative.cs
namespace BackendDotnet.Models;

public sealed class Initiative
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string Status { get; set; } = "pending";
    public string? BusinessProblem { get; set; }
    public string? ExpectedBenefit { get; set; }
    public DateTime CreatedAt { get; set; }
    public string? AnalysisResult { get; set; }
}

File: backend-fastapi/app/models/initiative.py
from datetime import datetime, timezone
from typing import Any

from sqlalchemy import JSON, DateTime, String, Text, func
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base


class Initiative(Base):
    __tablename__ = "initiatives"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    description: Mapped[str] = mapped_column(Text, nullable=False)
    status: Mapped[str] = mapped_column(String(50), nullable=False, default="pending")
    business_problem: Mapped[str | None] = mapped_column(Text, nullable=True)
    expected_benefit: Mapped[str | None] = mapped_column(Text, nullable=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        nullable=False,
        default=lambda: datetime.now(timezone.utc),
        server_default=func.now(),
    )
    analysis_result: Mapped[dict[str, Any] | None] = mapped_column(JSON, nullable=True)
